src/webots/nodes/utils/WbWorld.cpp

```diff
@@ -432,6 +432,7 @@ void WbWorld::createX3DMetaFile(const QString &filename) const {
           deviceObject.insert("minPosition", motor->minPosition());
           deviceObject.insert("maxPosition", motor->maxPosition());
           deviceObject.insert("position", motor->position());
+          deviceObject.insert("initialPosition", motor->joint()->solidEndPoint()->rotation().angle());
```

Comment:
I am fearing we are adding redundant information here. The previous version of the robot web component didn't have this information and could nevertheless display the range of motors motion properly.

Reply:
This may look as an issue, but I have done this on purpose. If this value is not given, we have to mathematically retrieve it from the threejs object3d rotation. Aside from the fact that this conversion is complex and time consuming, this operation is also certainly impossible: I think some information is lost during the load because of the eulerAxisAndAngle to quaternion implying that the inverse operation is not equivalent. It was possible with X3Dom because they probably have another rotation system closer from our rotation system.

I think the current way is much simpler, efficient and robust.

Reply:
I am sure it is not impossible to compute because we could compute it in the previous version. Maybe we should not compute it from the threejs object3d rotation, but earlier, from the x3d data we receive? Also, the computation should be done only once at load time and therefore it's performance impact is low.

Reply:
As told above, I think that the eulerAxisAndAngle->quaternion->eulerAxisAndAngle is not giving the same result, so doing as before is impossible.

It's indeed certainly to store the transform information in x3d.js during the load in the userData. But in this case I will pollute every other threejs projects with unneeded information. What I did here has the advantage to only have this information for the robot components which is the only location for now where we need this.

Reply:
My idea was not to store the transform information for every node in x3d.js during the load in the userData, but rather to compute the initialPosition of the motor (or whatever value we need) at this point.

Reply:
Computing the initial position of the motor after the threejs load is not possible because some information is lost.

Reply:
All right. I didn't noticed this information was in the robot web component specific x3dmeta file. Given it's specific to the web component, I believe this redundancy is acceptable if it simplifies the design.

Reply:
Ok thank you